README.md

```diff
@@ -1,111 +1,108 @@
+_out/node2:
+eth0.nmconnection  eth1.nmconnection
+
+_out/node3:
+eth2.nmconnection
 ```
 
-## How to run it?
+There are separate directories for each host (identified by their input <i>hostname</i>.yaml).
+Each of these contains the configuration files for the desired network interfaces (e.g. `eth0`).
 
-Using an example configuration of three known nodes (with hostnames `node1.example.com`, `node2.example.com`
-and `node3.example.com` and their respective NetworkManager settings) and interface mapping defined in `host_config.yaml`:
-
-```text
-config
-├── node1.example.com
-│   ├── eth0.nmconnection
-│   └── eth1.nmconnection
-├── node2.example.com
-│   └── eth0.nmconnection
-├── node3.example.com
-│   ├── bond0.nmconnection
-│   └── eth1.nmconnection
-└── host_config.yaml
-```
+The `host_config.yaml` file on the root level maps the hosts to all of their preconfigured interfaces.
+This is necessary in order for NMC to identify which host it is running on when applying the network configurations later.
 
-```shell
-$ ./nm-configurator -config-dir=config -hosts-config-file=host_config.yaml
-INFO[2023-08-17T17:32:23+03:00] starting network manager configurator...
-INFO[2023-08-17T17:32:23+03:00] successfully identified host: node1.example.com
-INFO[2023-08-17T17:32:23+03:00] storing file /etc/NetworkManager/system-connections/eth0.nmconnection... 
-INFO[2023-08-17T17:32:23+03:00] storing file /etc/NetworkManager/system-connections/eth1.nmconnection... 
-INFO[2023-08-17T17:32:23+03:00] successfully configured network manager 
+```yaml
+- hostname: node1
+  interfaces:
+    - logical_name: eth0
+      mac_address: FE:C4:05:42:8B:AA
+    - logical_name: bond0
+      mac_address: 0E:4D:C6:B8:C4:72
+...
 ```
 
-*Note:* The default values for `-config-dir` and `-hosts-config-file` flags are `config` and `host_config.yaml`
-respectively so providing them is not necessary with the file structure in the example:
+### Apply config
+
+NMC will use the previously generated configurations to identify and store the relevant
+NetworkManager settings for a given host out of the pool of predefined desired states.
+
+Typically used with [Combustion](https://documentation.suse.com/sle-micro/5.5/single-html/SLE-Micro-deployment/#cha-images-combustion)
+in order to bootstrap multiple nodes using the same provisioning artefact instead of depending on different custom images per machine.
 
 ```shell
-$ ./nm-configurator
-INFO[2023-08-17T17:45:41+03:00] starting network manager configurator...
-INFO[2023-08-17T17:45:41+03:00] successfully identified host: node1.example.com
-INFO[2023-08-17T17:45:41+03:00] storing file /etc/NetworkManager/system-connections/eth0.nmconnection... 
-INFO[2023-08-17T17:45:41+03:00] storing file /etc/NetworkManager/system-connections/eth1.nmconnection... 
-INFO[2023-08-17T17:45:41+03:00] successfully configured network manager 
+$ export RUST_LOG=info # set log level ("error" by default)
```

Comment:
Wouldn't be nice to have a `-v` or `--log-level` flag instead of this? Not something urgent, just thinking out loud.

Reply:
Will address in a follow up and adjust the docs once it's done.